agent/src/main/java/com/jonnyzzz/teamcity/plugins/node/agent/NodeDetector.kt

```diff
@@ -40,7 +40,7 @@ class NodeToolsDetector(events: EventDispatcher<AgentLifeCycleListener>,
   fun detectNVMTool() {
     with(config.systemInfo) {
       when {
-        isWindows && !File(System.getenv("APPDATA") + "\\nvm\\nvm.exe").isFile -> {
+        isWindows && !File(System.getenv("NVM_HOME") + "\\nvm.exe").isFile -> {
```

Comment:
It looks like a breaking change. Why don't check both to preserve backward compatibility?

Also, the `System.getenv("NVM_HOME") + "\\nvm.exe"` may turn to `null/nvm.exe` and it may even exist on the working directory. The code like `System.getenv("NVM_HOME")?.let { File(it, "nvm.exe") }` is a possible expression alternative (may be another implementataions too of course)

Reply:
Good point on the breaking change. I have factored it back in, and took on your advice about the null check with the env var.